web/src/pages/Setting/Setting.tsx

```diff
@@ -62,7 +62,20 @@ const Setting: React.FC = () => {
         });
         setTimeout(() => {
           const redirect = getUrlQuery('redirect');
-          window.location.href = redirect ? decodeURIComponent(redirect) : '/';
+          const currentHost = window.location.host;
+          if (redirect) {
+            const redirectUrl = decodeURIComponent(redirect);
+            const pathArray = redirectUrl.split('/');
+            const redirectHost = pathArray[2];
+            if (currentHost === redirectHost) {
+              let path = '';
+              for (let i = 3; i < pathArray.length; i++) {
+                path += '/';
+                path += pathArray[i];
+              }
+              history.push(path);
+            }
+          }
```

Comment:
it still needs a case to cover when redirect is ''
```suggestion
          } else {
          history.push('/');
          }
```

Reply:
Okay, I'll commit sugesstion